types/types.go

```diff
@@ -273,6 +273,14 @@ func (e *NotFoundError) Error() string {
 	return fmt.Sprintf("cannot find %v", e.Name)
 }
 
+type ErrorNotWithAttached struct {
+	Reason string
+}
+
+func (e *ErrorNotWithAttached) Error() string {
+	return fmt.Sprintf("current attached state prevents this: %v", e.Reason)
+}
```

Comment:
Suggest making this more generic, as there is already a specific reason provided.

```suggestion
func (e *ErrorInvalidState) Error() string {
	return fmt.Sprintf("invalid state: %v", e.Reason)
}
```